### ISSUE TITLE
Structuredata file is way too big

### ISSUE THREAD

**Issue by DBotThePony** [Ivorforce/RecurrentComplex#353]
*This is an long term issue, but it should be resolved somewhere in future*

Eventually, saving structuredata for any reason will become a bottleneck in server's work, because file will grow >50 mb, casuing huge lag (>10 seconds) when Recurrent Complex save it's data (sometimes lag even bigger than jungle maze generation). Disk is not a bottleneck - server (Linux based) runs on SSD.

Edit: Maybe store dat files per region? (32x32 chunks)

**Comment by Zekrom64** [comment]
Would second this, helping run a big server and tracked down a regular crash happening due to the server running out of memory trying to read a 161MB structuredata file. When I tried opening it in NBTExplorer it was well over 4GB uncompressed.

**Comment by Ivorforce** [comment]
Yep, per-region (or per-structure? hrm) storing is the way to go. I will tackle this some point after the other issues are through.

**Comment by Ivorforce** [comment]
Workaround btw is deleting the structureData file. It contains stuff that's important for structure generation and can mean structures at the edges of your world are cut off, and /#whatisthis will fail on old structures, but otherwise it has no real downsides.
